File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/adapters/ShortcutListAdapter.java
package ch.rmy.android.http_shortcuts.adapters;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.Bind;
import butterknife.ButterKnife;
import ch.rmy.android.http_shortcuts.R;
import ch.rmy.android.http_shortcuts.icons.IconView;
import ch.rmy.android.http_shortcuts.realm.models.PendingExecution;
import ch.rmy.android.http_shortcuts.realm.models.Shortcut;

public class ShortcutListAdapter extends ShortcutAdapter {

    public ShortcutListAdapter(Context context) {
        super(context);
    }

    @Override
    protected ShortcutViewHolder createViewHolder(ViewGroup parentView) {
        return new ShortcutViewHolder(parentView);
    }

    public class ShortcutViewHolder extends BaseViewHolder<Shortcut> {

        @Bind(R.id.name)
        TextView name;
        @Bind(R.id.description)
        TextView description;
        @Bind(R.id.icon)
        IconView icon;
        @Bind(R.id.waiting_icon)
        View waitingIcon;

        public ShortcutViewHolder(ViewGroup parent) {
            super(LayoutInflater.from(context).inflate(R.layout.shortcut_list_item, parent, false), ShortcutListAdapter.this);
            ButterKnife.bind(this, itemView);
        }

        @Override
        protected void updateViews(Shortcut shortcut) {
            name.setText(shortcut.getName());
            description.setText(shortcut.getDescription());
            description.setVisibility(TextUtils.isEmpty(shortcut.getDescription()) ? View.GONE : View.VISIBLE);
            icon.setImageURI(shortcut.getIconURI(context), shortcut.getIconName());
            waitingIcon.setVisibility(isPendingExecution(shortcut.getId()) ? View.VISIBLE : View.GONE);
        }

        private boolean isPendingExecution(long shortcutId) {
            for (PendingExecution pendingExecution : shortcutsPendingExecution) {
                if (pendingExecution.getShortcutId() == shortcutId) {
                    return true;
                }
            }
            return false;
        }

    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/adapters/ShortcutListAdapter.kt
package ch.rmy.android.http_shortcuts.adapters

import android.content.Context
import android.text.TextUtils
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import ch.rmy.android.http_shortcuts.R
import ch.rmy.android.http_shortcuts.icons.IconView
import ch.rmy.android.http_shortcuts.realm.models.Shortcut

class ShortcutListAdapter(context: Context) : ShortcutAdapter(context) {

    override fun createViewHolder(parentView: ViewGroup) = ShortcutViewHolder(parentView)

    inner class ShortcutViewHolder(parent: ViewGroup) : BaseViewHolder<Shortcut>(parent, R.layout.shortcut_list_item, this@ShortcutListAdapter) {

        internal var name = itemView.findViewById(R.id.name) as TextView
        internal var description = itemView.findViewById(R.id.description) as TextView
        internal var icon = itemView.findViewById(R.id.icon) as IconView
        internal var waitingIcon = itemView.findViewById(R.id.waiting_icon)

        override fun updateViews(item: Shortcut) {
            name.text = item.name
            description.text = item.description
            description.visibility = if (TextUtils.isEmpty(item.description)) View.GONE else View.VISIBLE
            icon.setImageURI(item.getIconURI(context), item.iconName)
            waitingIcon.visibility = if (isPendingExecution(item.id)) View.VISIBLE else View.GONE
        }

        private fun isPendingExecution(shortcutId: Long) = shortcutsPendingExecution!!.any {
            it.shortcutId == shortcutId
        }

    }

}
